TeslaMateAgile/Services/BarryService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Microsoft.Extensions.Options;
+using System.Globalization;
+using System.Text.Json.Serialization;
+using TeslaMateAgile.Data;
+using TeslaMateAgile.Data.Options;
+using TeslaMateAgile.Services.Interfaces;
+
+namespace TeslaMateAgile.Services;
+
+public class BarryService : IPriceDataService
+{
+    private readonly HttpClient _client;
+    private readonly BarryOptions _options;
+
+    public BarryService(HttpClient client, IOptions<BarryOptions> options)
+    {
+        _client = client;
+        _options = options.Value;
+    }
+
+    public async Task<IEnumerable<Price>> GetPriceData(DateTimeOffset from, DateTimeOffset to)
+    {
+        BarryRequest request = new BarryRequest {
+            Params = new String[] { _options.BarryMPID, from.UtcDateTime.ToString("o"), to.UtcDateTime.ToString("o")}.ToList()
+        };
+        var objAsJson = System.Text.Json.JsonSerializer.Serialize<BarryRequest>(request);
+        var content = new StringContent(objAsJson, System.Text.Encoding.UTF8, "application/json");
+
+        var resp = await _client.PostAsync(_options.BaseUrl, content);
+        resp.EnsureSuccessStatusCode();
+
+        var agileResponse = await System.Text.Json.JsonSerializer.DeserializeAsync<BarryResponse>(await resp.Content.ReadAsStreamAsync());
+        
+        return agileResponse.Results.Select(x => new Price
+        {
+            Value = x.Value,
+            ValidFrom = BarryService.ParseIso8601(x.Start),
+            ValidTo = BarryService.ParseIso8601(x.End)
```

Comment:
I believe you shouldn't have to reference BarryService here but keep this in if it is required

```suggestion
            ValidFrom = ParseIso8601(x.Start),
            ValidTo = ParseIso8601(x.End)
```